pkg/apis/ceph.rook.io/v1/types.go

```diff
@@ -3265,6 +3265,33 @@ type CephBlockPoolRadosNamespaceList struct {
 	Items           []CephBlockPoolRadosNamespace `json:"items"`
 }
 
+// RadosNamespaceMirroring represents the mirroring configuration of CephBlockPoolRadosNamespaceSpec
+type RadosNamespaceMirroring struct {
+	// Enabled is a flag to enable or disable mirroring
+	// +optional
+	Enabled bool `json:"enabled"`
+	// RemoteNamespace is the name of the radosNamespace on secondary cluster pool
+	// +optional
+	RemoteNamespace string `json:"remoteNamespace"`
+	// Mode is the mirroring mode; either pool or image
+	// +optional
+	// +kubebuilder:validation:Enum=pool;image
+	Mode RadosNamespaceMirroringMode `json:"mode"`
+	// SnapshotSchedules is the scheduling of snapshot for mirrored images
+	// +optional
+	SnapshotSchedules []SnapshotScheduleSpec `json:"snapshotSchedules,omitempty"`
+}
+
+// RadosNamespaceMirroringMode represents the mode of the RadosNamespace
+type RadosNamespaceMirroringMode string
+
+const (
+	// RadosNamespaceMirroringModePool represents the pool mode
+	RadosNamespaceMirroringModePool RadosNamespaceMirroringMode = "pool"
+	// RadosNamespaceMirroringModeImage represents the image mode
+	RadosNamespaceMirroringModeImage RadosNamespaceMirroringMode = "image"
+)
```

Comment:
We have the same mirroring modes for cephblockpool as well, should also consider making the cephblockpool.spec.mirroring as an enum and have a predefined const for it as well? If not in this PR maybe in a new PR?

Reply:
Yaa have plan to update blockpool with a follow up PR, there are lot of gaps